app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -170,9 +177,32 @@ def process_collection(collection, key)
         end
       end
 
-      def gather_data_for_this_region(arm_service, method = "list")
-        @data[:resource_groups].collect do |resource_group|
-          arm_service.send(method, resource_group[:name])
+      # Gather all data for all resource groups. If a subscription ID is provided,
+      # then only collect data for that subscription. Otherwise, collect all data
+      # for all subscriptions.
+      #
+      def gather_data_for_this_region(arm_service, method = 'list_all', subscription_id = nil)
+        subscriptions = subscription_id ? [subscription_id] : @subscriptions
```

Comment:
@bronaghs Just looking at this again, we're going to have to replace all instances of @subscription_id in the various parse methods. Instead we'll need to parse the subscription id out of the object's id method.